inc/admin/options-page.php

```diff
@@ -187,6 +211,29 @@ public function field_text( $args ) {
 	}
 
 	/**
+	 * Output a checkbox.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param  array  $args  The array of field args.
+	 */
+	public function field_checkbox( $args ) {
+
+		$option_id          = 'alcatraz-options-' . str_replace( '_', '-', $args['id'] );
+		$option_key         = 'alcatraz_options[' . $args['id'] . ']';
+		$option_value       = alcatraz_true_or_false( $this->options[ $args['id'] ] );
+		$option_description = ( ! empty( $args['description'] ) ) ? '<br /><span class="description">' . wp_kses_post( $args['description'] ) . '</span>' : '';
+
+		printf(
+			'<input type="checkbox" class="checkbox" id="%s" name="%s" value="1" %s />%s',
+			esc_attr( $option_id ),
+			esc_attr( $option_key ),
+			checked( $option_value, 1, false ),
```

Comment:
The key to the issue is here. You are doing this the correct way, but this will either submit the value as 1 or it will not submit a value (when it is unchecked). Right now in our validation function we check for `if ! empty()`, but if the box isn't checked then that key on $input won't be set at all, so it's currently impossible to set the option back to 0.

This is related to the fact that we're validating both our options page options and Customizer options in a single function. When we submit from the options page, the $input param going into our validation function only at most contains the fields on the options page, and when we submit from the Customizer, the $input param only contains our Customizer options.

If we didn't have to worry about the Customizer scenario, we could simply say if the key 'social_icons_in_footer' isn't equal to 1 then set it to 0, but if we did this now then every time we save in the Customizer we would reset the option to 0.

One way we can solve this problem is to detect whether $input is coming from the options page or the Customizer, and set it to 0 if it isn't set to 1 if it's coming from the options page. There are several ways we could potentially detect that $input is coming from the options page, like looking at the $_POST and $_SERVER globals, looking for the presence of other options that are always there, setting a hidden input, etc.

But I actually think this option should live in the Customizer. It's a visual change that users would notice taking effect as they change it, and we already have a Footer section in the Customizer. I think we should add a big blue link below the social network URL inputs on the options page that says something like "View all design options in the Customizer".

I could be wrong, but I think if we have the option live in the Customizer and use the standard Customizer control for checkboxes, we'll get both the 0 and the 1 passed when the user saves, so our standard `! empty()` check will work. That doesn't fix the issue with using `field_checkbox()` on our options page, so I'll take more of a look at how we can solve that problem when I get a chance, or if you have an idea feel free to try it out. Ultimately I think this option should live in the Customizer but we should have a working field_checkbox in our options page class also.